instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/datasource/JdbcTelemetry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jdbc.datasource;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.jdbc.internal.DbRequest;
+import io.opentelemetry.instrumentation.jdbc.internal.dbinfo.DbInfo;
+import javax.sql.DataSource;
+
+/** Entrypoint for instrumenting a JDBC DataSources. */
+public class JdbcTelemetry {
+
+  /** Returns a new {@link JdbcTelemetry} configured with the given {@link OpenTelemetry}. */
+  public static JdbcTelemetry create(OpenTelemetry openTelemetry) {
+    return builder(openTelemetry).build();
+  }
+
+  /** Returns a new {@link JdbcTelemetryBuilder} configured with the given {@link OpenTelemetry}. */
+  public static JdbcTelemetryBuilder builder(OpenTelemetry openTelemetry) {
+    return new JdbcTelemetryBuilder(openTelemetry);
+  }
+
+  private final Instrumenter<DataSource, DbInfo> dataSourceInstrumenter;
+  private final Instrumenter<DbRequest, Void> statementInstrumenter;
+
+  protected JdbcTelemetry(
+      Instrumenter<DataSource, DbInfo> dataSourceInstrumenter,
+      Instrumenter<DbRequest, Void> statementInstrumenter) {
+    this.dataSourceInstrumenter = dataSourceInstrumenter;
+    this.statementInstrumenter = statementInstrumenter;
+  }
+
+  public OpenTelemetryDataSource wrap(DataSource dataSource) {
```

Comment:
Would it make more sense to just return `DataSource` here?

Reply:
I've updated the PR to return `DataSource` instead of the concrete type `OpenTelemetryDataSource`. I can't find of a reason why this isn't better.